### PR TITLE
Add support for building rootfs.ext2

### PR DESCRIPTION
**Pull request by vfusco**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,101 +15,83 @@
...
-	cat tools/template/control.template | sed 's|ARG_VERSION|$(VERSION)|g' > $(DESTDIR)/DEBIAN/control
-	cp -R $(STAGING_BASE)* $(DESTDIR)$(PREFIX)
-	dpkg-deb -Zxz --root-owner-group --build $(DESTDIR) $(MACHINE_EMULATOR_TOOLS_DEB)
-
-$(LINUX_SOURCES_FILEPATH):
-	T=`mktemp` && \
-	    mkdir -p `dirname $@` &&  \
-	    wget $(LINUX_SOURCES_URLPATH) -O $$T && \
-	    mv $$T $@ || (rm $$T && false)
+$(TOOLS_DEB) deb: build
```
Comment (mpolitzer) [inline Makefile:51]: I find this clearer.
```suggestion
deb: $(TOOLS_DEB)
$(TOOLS_DEB): build
```

File: Makefile [hunk 2]
```diff
@@ -15,101 +15,83 @@
...
-	@echo '  distclean       - clean and remove dependencies'
 
-clean:
-	rm -f $(MACHINE_EMULATOR_TOOLS_TAR_GZ) $(MACHINE_EMULATOR_TOOLS_DEB)
+clean-image:
+	@(docker rmi $(TOOLS_IMAGE) > /dev/null 2>&1 || true)
 
-distclean: clean
-	rm -rf dep
+distclean clean: clean-image
```
Comment (mpolitzer) [inline Makefile:94]: Will `clean` require rebuilding the docker images? I don't like it if that is the case.

File: fs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM --platform=linux/riscv64 riscv64/ubuntu:22.04
+ARG TOOLS_DEB=machine-emulator-tools-v0.14.0.deb
+ADD ${TOOLS_DEB} /tmp/
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+      busybox-static=1:1.30.1-7ubuntu3 \
```
Comment (mpolitzer) [inline fs/Dockerfile:6]: This is redundant now that the dependency was added to the control file.

File: rollup-http/rollup-init/rollup-init [hunk 4]
```diff
@@ -16,7 +16,7 @@
 # limitations under the License.
 #
 
-busybox sh -c "/opt/cartesi/bin/rollup-http-server $*"
+busybox sh -c "/usr/bin/rollup-http-server $*"
```
Comment (mpolitzer) [inline rollup-http/rollup-init/rollup-init:19]: If we don't namespace our tools anymore, do we still need full paths?

File: README.md [hunk 5]
```diff
@@ -1,39 +1,46 @@
 # Cartesi Machine Emulator Tools
 
-The Cartesi Machine Emulator Tools a repository that contains a set of tools developed for the RISC-V Linux OS.
+The Cartesi Machine Emulator Tools is a repository that contains a set of tools developed for the RISC-V Linux OS. It provides a RISC-V Debian package and a root filesystem for Ubuntu 22.04.
 
 ## Getting Started
 
-Users looking to create cartesi-machine applications can use the artifact directly, no need to build this repository themselves. Also check the `example/` folder for the details.
+Users looking to create cartesi-machine applications can use the provided Debian package and root filesystem directly, without needing to build this repository themselves.
```
Comment (mpolitzer) [inline README.md:7]:
```suggestion
Users looking to create cartesi-machine applications can use the provided Debian package and root filesystem directly from one of the prepackaged [releases](https://github.com/cartesi/machine-emulator-tools/releases), without needing to build this repository themselves.
```

File: Makefile [hunk 6]
```diff
@@ -15,101 +15,83 @@
...
+			--build-arg IMAGE_KERNEL_VERSION=$(IMAGE_KERNEL_VERSION) \
+			--build-arg LINUX_VERSION=$(LINUX_VERSION) \
+			--build-arg LINUX_HEADERS_URLPATH=$(LINUX_HEADERS_URLPATH) \
+			-t $(TOOLS_IMAGE) \
+			-f Dockerfile \
+			. ; \
+	fi
+	@$(MAKE) copy
 
 copy:
```
Comment (mpolitzer) [inline Makefile:46]:
```suggestion
copy: build
```

File: Makefile [hunk 7]
```diff
@@ -15,101 +15,83 @@
...
+		docker buildx build --platform=linux/riscv64 --load \
+			--build-arg TOOLS_DEB=$(TOOLS_DEB) \
+			--build-arg IMAGE_KERNEL_VERSION=$(IMAGE_KERNEL_VERSION) \
+			--build-arg LINUX_VERSION=$(LINUX_VERSION) \
+			--build-arg LINUX_HEADERS_URLPATH=$(LINUX_HEADERS_URLPATH) \
+			-t $(TOOLS_IMAGE) \
+			-f Dockerfile \
+			. ; \
+	fi
+	@$(MAKE) copy
```
Comment (mpolitzer) [inline Makefile:44]:
```suggestion
```

File: Makefile [hunk 8]
```diff
@@ -15,101 +15,83 @@
...
+TOOLS_IMAGE   := cartesi/machine-emulator-tools:$(VERSION)
+TOOLS_ROOTFS  := rootfs-tools-v$(VERSION).ext2
+
+IMAGE_KERNEL_VERSION ?= v0.19.0
+LINUX_VERSION ?= 6.5.9-ctsi-1
+LINUX_HEADERS_URLPATH := https://github.com/cartesi/image-kernel/releases/download/${IMAGE_KERNEL_VERSION}/linux-libc-dev-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
+
+all: fs
+
+build: control
```
Comment (mpolitzer) [inline Makefile:33]:
```suggestion
build-image: control
```

File: Makefile [hunk 9]
```diff
@@ -15,101 +15,83 @@
...
-	    wget $(LINUX_SOURCES_URLPATH) -O $$T && \
-	    mv $$T $@ || (rm $$T && false)
+$(TOOLS_DEB) deb: build
 
-$(RNDADDENTROPY_FILEPATH):
-	T=`mktemp` && \
-	    mkdir -p `dirname $@` &&  \
-	    wget $(RNDADDENTROPY_URLPATH) -O $$T && \
-	    mv $$T $@ || (rm $$T && false)
+control:
```
Comment (mpolitzer) [inline Makefile:53]:
```suggestion
control: Makefile
```

File: .github/workflows/main.yml [hunk 10]
```diff
@@ -5,11 +5,25 @@ env:
...
-        with:
-          submodules: recursive
+      - name: Install libarchive-tools
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends build-essential autoconf automake libarchive-dev libarchive-tools
+
+      - name: Download, build and install xgenext2fs
+        run: |
```
Comment (mpolitzer) [inline .github/workflows/main.yml:17]: Because `.deb` from genext2fs repo builds for `debian` and not `ubuntu`?
If so, I think this deserves a comment.

**Review by mpolitzer — COMMENTED** [review]
I'll create put the comments on Makefile on a fixup, in a separate branch.
Suggestions get too confusing here.

File: Dockerfile [hunk 11]
```diff
@@ -14,79 +14,125 @@
...
-      protobuf-compiler \
-      rsync \
-      rust-all=1.58.1+dfsg1~ubuntu1-0ubuntu2 \
+      wget \
+      crossbuild-essential-riscv64 \
+      gcc-12-riscv64-linux-gnu \
+      g++-12-riscv64-linux-gnu \
       && \
-    rm -rf /var/lib/apt/lists/*
+    wget -O ${LINUX_HEADERS_FILEPATH} ${LINUX_HEADERS_URLPATH} && \
```
Comment (mpolitzer) [inline Dockerfile:38]: As a simplification:
We could use Dockers `ADD` command to fetch this file instead, move the `sha256sum` at the top of this `RUN` and apt-get install everything at once.

File: rollup-http/rollup-init/rollup-init (hunk 4 above)
Comment (vfusco) [reply rollup-http/rollup-init/rollup-init:19]: Maybe we can change this in the future, but for now I think explicit is better.

**Review by mpolitzer — APPROVED** [review]

**Pull request merged**